spec/models/relationship_spec.rb

```diff
@@ -91,4 +91,18 @@
       expect(filtered_results).to match_array(vms + hosts)
     end
   end
+
+  describe ".filtered" do
+    let(:storages) { FactoryGirl.build_list(:relationship_storage_vmware, 1) }
+    let(:vms) { FactoryGirl.build_list(:relationship_vm_vmware, 1) }
+    let(:hosts) { FactoryGirl.build_list(:relationship_host_vmware, 1) }
+
+    it "scopes" do
+      vms.map(&:save!)
+      hosts.map(&:save!)
+      storages.map(&:save!)
+      filtered_results = Relationship.filtered(%w(Host VmOrTemplate), %w(Storage))
```

Comment:
I've been avoiding `let!` - want to be obvious in the test what data exists.
the `!` is so subtle and makes this hidden. It also creates objects and people refactoring don't know if it is necessary or if someone was just paying attention when adding a test.

Removing the let!(:storage) would still pass the tests, but it would not be testing that `storages` are not being brought back. So the test would become useless.